test/test-server/provider_integration_handler.go

```diff
@@ -0,0 +1,171 @@
+				Config:      configs[0],
+				Usages:      &[]string{},
+			},
+			{
+				Id:          pi.PtrString("cspi-4xgwq"),
+				DisplayName: pi.PtrString("cdong-test"),
+				Provider:    pi.PtrString("aws"),
+				Environment: &pi.GlobalObjectReference{Id: "env-9zgy77"},
+				Config:      configs[1],
+				Usages:      &[]string{},
+			},
+		}
+		result := pi.PimV1IntegrationList{Data: mockResponse}
+		err := json.NewEncoder(w).Encode(result)
+		require.NoError(t, err)
+	}
+}
+
+func handleProviderIntegrationCreate(t *testing.T) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var mockRequest pi.PimV1Integration
+		require.NoError(t, json.NewDecoder(r.Body).Decode(&mockRequest))
+
+		provider := mockRequest.GetProvider()
+		mockResponse := &pi.PimV1Integration{
+			Id:          pi.PtrString("cspi-42o61"),
+			DisplayName: pi.PtrString("cdong-test2"),
+			Provider:    pi.PtrString(provider),
+			Environment: &pi.GlobalObjectReference{Id: "env-9zgy77"},
+			Config:      getPimV1IntegrationConfig(provider),
+		}
+
+		err := json.NewEncoder(w).Encode(mockResponse)
+		require.NoError(t, err)
+	}
+}
+
+// Return provider specific config, Azure and Gcp are to be implemented from SDK
+func getPimV1IntegrationConfig(provider string) *pi.PimV1IntegrationConfigOneOf {
+	provider = strings.ToLower(provider)
+	switch provider {
+	case "aws":
+		return &pi.PimV1IntegrationConfigOneOf{
+			PimV1AwsIntegrationConfig: &pi.PimV1AwsIntegrationConfig{
+				IamRoleArn:         pi.PtrString("arn:aws:iam::851725421142:role/cspi-42o61"),
+				ExternalId:         pi.PtrString("999219d4-37f4-49ac-abfe-d2b6528fb21b"),
+				CustomerIamRoleArn: pi.PtrString("arn:aws:iam::037803949979:role/tarun-iam-test-role"),
+				Kind:               AwsIntegrationConfig,
+			},
+		}
+	default:
+		return &pi.PimV1IntegrationConfigOneOf{}
+	}
+}
+
+// Return provider specific config list, Azure and Gcp are to be implemented from SDK
+func getPimV1IntegrationConfigList(provider string) []*pi.PimV1IntegrationConfigOneOf {
+	provider = strings.ToLower(provider)
+	switch provider {
+	case "aws":
+		return []*pi.PimV1IntegrationConfigOneOf{
+			{
+				PimV1AwsIntegrationConfig: &pi.PimV1AwsIntegrationConfig{
+					IamRoleArn:         pi.PtrString("arn:aws:iam::851725421142:role/cspi-42o61"),
+					ExternalId:         pi.PtrString("999219d4-37f4-49ac-abfe-d2b6528fb21b"),
+					CustomerIamRoleArn: pi.PtrString("arn:aws:iam::037803949979:role/tarun-iam-test-role"),
+					Kind:               AwsIntegrationConfig,
+				},
+			},
+			{
+				PimV1AwsIntegrationConfig: &pi.PimV1AwsIntegrationConfig{
+					IamRoleArn:         pi.PtrString("arn:aws:iam::851725421142:role/cspi-4xgwq"),
+					ExternalId:         pi.PtrString("b5f0a7e6-15e3-4879-aa01-881910554cd4"),
+					CustomerIamRoleArn: pi.PtrString("arn:aws:iam::000000000000:role/my-test-aws-role"),
+					Kind:               AwsIntegrationConfig,
+				},
+			},
+		}
+	default:
+		return []*pi.PimV1IntegrationConfigOneOf{}
```

Comment:
Does this line ever get reached? I'm not sure we need it.

Reply:
It's never reached, my original intention is to have a placeholder for the upcoming cloud providers.

I just removed the redundant part to keep it clean.